ale_linters/perl/perl.vim

```diff
@@ -27,12 +27,20 @@ function! ale_linters#perl#perl#Handle(buffer, lines) abort
     let l:output = []
     let l:basename = expand('#' . a:buffer . ':t')
 
+    let type = 'E'
+    let syntax_ok = matchstr(a:lines[-1], 'syntax OK')
```

Comment:
If you just want to check if there is a match, you can use the operator for that. See `:help =~#`.

Reply:
I also prohibit the use of `==#` and `==?` from the codebase, as they behave poorly when you mix numbers and strings. `is#` and is?` are preferred instead.

Reply:
Fixed in f66d38bab7